src/ports/nfts/utils.ts

```diff
@@ -257,7 +257,7 @@ export function getFetchQuery(
       ${getExtraWhere ? getExtraWhere(filters).join('\n') : ''}
     }`
 
-  if (wrapWhere) {
+  if (wrapWhere && filters.caller !== filters.owner) {
```

Comment:
By doing this, if the caller === owner, no queries the `searchOrderExpiresAt` will never be used for the query, meaning it will not fetch by isOnSale